DatadogSessionReplay/Sources/DatadogSessionReplay/Utilities/UIImage+Compress.swift

```diff
@@ -0,0 +1,37 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+
+extension UIImage {
+    func compressToTargetSize(_ targetSize: Int) -> Data? { // you can know average needed scale
+        var compressionQuality: CGFloat = 1.0
+        guard var imageData = pngData() else {
+            return nil
+        }
+        guard imageData.count >= targetSize else {
+            return imageData
+        }
+        var image = self
+        while imageData.count > targetSize {
+            compressionQuality -= 0.1
+            imageData = image.jpegData(compressionQuality: compressionQuality)!
+
+            if imageData.count > targetSize {
+                image = image.scaledImage(by: 0.9)
+            }
+        }
```

Comment:
This loop can be costly, maybe we can compute the quality and scale based on the data size and target?

Reply:
I tried different implementations for that one and nothing worked as nicely. I'll create a ticket to revisit that.